server/src/events.mjs

```diff
@@ -87,70 +100,152 @@ function getRegisteredEventListener(userId, method) {
   return eventListenerMap[userId][method];
 }
 
-// Remove mapping from event listener request method name from our map
-// Attempts to send events to this listener going forward will fail
-function deregisterEventListener(userId, oMsg, ws) {
-  if (!eventListenerMap[userId] || !eventListenerMap[userId][oMsg.method]) {
+/**
+ * Removes the mapping from the event listener request method name from eventListenerMap. 
+ * Attempts to send events to this listener going forward will fail.
+ * @param {string} userId - The ID of the user.
+ * @param {Object} metadata - An object containing metadata for the event.
+ * @param {WebSocket} ws - The WebSocket object.
+ * @returns {void}
+*/
+function deregisterEventListener(userId, metadata, ws) {
+  const { method } = metadata;
+  if (!eventListenerMap[userId] || !eventListenerMap[userId][method]) {
     return;
   }
 
-  const wsArr = eventListenerMap[userId][oMsg.method].wsArr;
+  const wsArr = eventListenerMap[userId][method].wsArr;
   const wsIndex = wsArr.findIndex((item) => item === ws);
 
   if (wsIndex !== -1) {
     wsArr.splice(wsIndex, 1);
-    logger.debug(`Deregistered event listener mapping: ${userId}:${oMsg.method}`);
+    logger.debug(`Deregistered event listener mapping: ${userId}:${method}`);
   }
 
   if (wsArr.length === 0) {
-    delete eventListenerMap[userId][oMsg.method];
+    delete eventListenerMap[userId][method];
+  }
+}
+
+/**
+ * Extracts event data from a given message object based on provided configuration.
+ * @param {object} oMsg - The message object to extract event data from.
+ * @param {object} config - The configuration object for the event data extraction.
+ * @param {boolean} isOn - Whether eventListener enable or disable request
+ * @returns {object | false} - An object containing the extracted event data or false if the extraction fails.
+*/
+function extractEventData(oMsg, config, isOn) {
+  const searchRegex = config.searchRegex;
+  const method = config.method || '$.method';
+
+  if (!new RegExp(searchRegex).test(JSON.stringify(oMsg))) {
+    return false;
   }
+
+  const extractedMethod = JSONPath.query(oMsg, method);
+
+  if (extractedMethod.length === 0) {
```

Comment:
Can we add a log here to indicate why we're failing?

Reply:
Added following log

    logger.debug(`Error occurred while extracting event data: No method found in the provided message.`);